lib/streamlit/elements/widgets/data_editor.py

```diff
@@ -787,7 +791,24 @@ def data_editor(
         # Throws an exception if any of the configured types are incompatible.
         _check_type_compatibilities(data_df, column_config_mapping, dataframe_schema)
 
+        id = new_compute_widget_id(
```

Comment:
is this something that we want to do closer to the top of the method so that anything that happens in between can't possibly affect the ID? If not, we might want a comment explaining why we defer generating this ID until later